php_phongo.h

```diff
@@ -97,6 +97,12 @@ typedef enum {
  * or command should select ExecutionTimeoutException. */
 #define PHONGO_SERVER_ERROR_EXCEEDED_TIME_LIMIT 50
 
+/* These constants are used for determining if a server error is due to a
+ * broken cursor. If so, we select a ConnectionException as this is what we did
+ * when getMore wasn't a command yet. */
```

Comment:
If I understand correctly, a ConnectionException was previously thrown because we experienced an opcode failure instead of a mere command failure (with `ok: 0`). I understand the desire for consistency in error reporting, but are we lying to users here?

I would think that a ConnectionException implies that the socket itself is discarded. It's possible that libmongoc may very well have done that for the old getmore failure (I don't know offhand so it bears investigation), but I don't think it would for a failed command.

Both of these errors seem very much like ServerExceptions. Although both ConnectionException and ServerException extend RuntimeException (and may report error labels), users can now rest assured that these error codes come from the server, which makes ServerException more sensible.

With the older getmore opcode failures, there was no server-side error code, so ConnectionException (or a base RuntimeException) would have made sense.

Reply:
But this comes at a cost of inconsistency with regard to error code reporting. We tell users that that any server-originating error codes are used in exceptions within the ServerException hierarchy.

Am I correct in assuming that the error codes for killed cursors using OP_GETMORE differ from the command's error code? I believe so, as `MONGOC_ERROR_CURSOR_INVALID_CURSOR` was simply a sequential value in `mongoc_error_code_t` and that is what is used in `_mongoc_rpc_check_ok()` when inspecting an OP_REPLY. When the getMore command used, libmongoc now reports the error differently and uses the new server-originating error codes, which also changes how users can detect and differentiate each error.

IMO, the error reporting inconsistency is due to the server's API. Given that, I think the variation of exceptions is excusable based on the wire protocol we're using, and using ServerException is actually an improvement when it comes to distinguishing the errors via their code.

Reply:
I've removed this now, and split the two test cases in a pre-3.6 and post-3.6 version.